src/persist-client/src/impl/machine.rs

```diff
@@ -64,6 +64,14 @@ where
         self.state.shard_id()
     }
 
+    pub async fn upper(&mut self, deadline: Instant) -> Result<Antichain<T>, ExternalError> {
+        // This unconditionally fetches the latest state and uses that to
```

Comment:
copy-pasta? I'm not sure I see how this comment applies here

Reply:
yes 🙈